src/storage/config.rs

```diff
@@ -113,6 +109,15 @@ impl Config {
                     .into(),
             );
         };
+        let max_pool_size = std::cmp::max(1, SysQuota::cpu_cores_quota() as usize);
+        if self.scheduler_worker_pool_size == 0 || self.scheduler_worker_pool_size > max_pool_size {
```

Comment:
Not sure if it is a good idea, since TiKV uses block io currently, the user may want to use more threads to get full utilization of cpu cores. And since now we support changing the pool size online, the user can always decrease the pool size if it hurt performance, so may not need such restricted restraint.

Reply:
Because yatp thread pool need to set the `max-thread-count` before initializing, and after the pool is built, and dynamic pool size change can not exceed this threshold. And, bacause yatp pool need to pre-create `max-thread-count` threads, so I think the cpu_num should be a proper upper limit.  /cc @ethercflow

Reply:
> Because yatp thread pool need to set the max-thread-count before initializing, and after the pool is built, and dynamic pool size change can not exceed this threshold. And, bacause yatp pool need to pre-create max-thread-count threads, so I think the cpu_num should be a proper upper limit. /cc @ethercflow

For IDC environment, I think `max_thread_count` is `SysQuota::cpu_cores_quota()` maybe enough.
For cloud environment, eg: 4 core or 8 core,  I've no idea.  How much do we need at most in this type of scenario? @NingLin-P

Reply:
/cc @BusyJay

Reply:
I don't think we have ever configure the scheduler worker pool larger than the cpu cores. But the situation may not be valid for 4 cores or even 1 cores. And we may need to bench how it performs when too many threads are spawn even without being used.

Reply:
So, how about set a hard threshold of max-threshold-count when the total CPU Quota is relative small, e.g. 4, or a dynamic config like (CPUQuota  * 2)?

Reply:
@NingLin-P @BusyJay @ethercflow what do you think? Or what other benchmark do we need to do?

Reply:
according @BusyJay 's recommendation, set the min value of `max-thread-count` to 4 when the cpu quota is smaller.